samples/graph-app-installation-lifecycle/csharp/AppInstallation/Views/Home/Index.cshtml

```diff
@@ -0,0 +1,288 @@
+    <div class="modal-dialog" role="document">
+        <div class="modal-content">
+            <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                    <span aria-hidden="true">&times;</span>
+                </button>
+            </div>
+            <div class="modal-body" id="modalbody">
+                <span>Are you sure you want to delete this App? </span>
+            </div>
+            <div class="modal-footer">
+                <button type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</button>
+                <button type="button" class="btn btn-outline-danger" id="btndelApp" data-dismiss="modal">Delete</button>
+            </div>
+        </div>
+    </div>
+</div>
+<div class="modal fade" id="userAppModal" tabindex="-1" role="dialog" aria-labelledby="userAppModalLabel" aria-hidden="true">
+    <div class="modal-dialog" role="document">
+        <div class="modal-content">
+            <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                    <span aria-hidden="true">&times;</span>
+                </button>
+            </div>
+            <div class="modal-body" id="usermodalbody">
+                
+            </div>
+        </div>
+    </div>
+</div>
+<div>
+   <h5>List of Apps:</h5>
+        <table border="1" >
+            <thead>
+                <tr>
+               <td>Sr No.</td> 
+                <td>Name</td>
+                <td>Info</td>
+                <td>Update</td>
+                <td>Delete</td>
+                </tr>
+            </thead>
+            <tbody>
+                @{int Count = 1;}
+                @foreach (var item in Model.appList) {
+                    
+                <tr style="padding:1px">
+                 <td>@Count</td>
+                <td>@item.AppName</td>
+                <td><button type="button" class="btn btn-outline-success btngetApp" data-app-itemid="@item.AppId">Get App</button></td>
+                <td><button type="button" class="btn btn-outline-success btnUpgradeApp" data-app-itemid="@item.AppId">Update</button></td>
+                <td><button type="button" class="btn btn-outline-info btndeleteApp" data-app-itemid="@item.AppId" data-app-itemname="@item.AppName">Delete</button></td>
+                </tr>
+                Count = Count + 1;
+                }
+            </tbody>
+        </table>
+</div>
+
+<script type="text/javascript">
+    $(document).ready(function () {
+        microsoftTeams.initialize();
+        var teamAppInstallationId = '';
+
+            $(".btngetApp").on('click', function (e) {
+                microsoftTeams.getContext(function (context) {
+                    var tenantId = context.tid;
+                    var teamId = context.groupId;
```

Comment:
We are fetching context evetime, instead you can call it once and save the information in variable and reuse.

Reply:
Updated this comment